Disable hiding window while in `MINIMIZED`, `FULLSCREEN` or `PRESENTATION` state and ignore no-op visibility requests (#3109)
beeware/toga: changes/3109.removal.rst

> Hiding of a window while in `MINIMIZED`, `FULLSCREEN` or `PRESENTATION` state is now disabled, in order to keep the API behavior consistent on all platforms.

Reviewer: ```suggestion
The `show()` and `hide()` APIs can no longer be used on a window while it is in a `MINIMIZED`, `FULLSCREEN` or `PRESENTATION` state.
```

Author: I think we do not need to mention `show()` here, since calling `window.show()` when in `MINIMIZED`, `FULLSCREEN` or `PRESENTATION` state, will be a no-op, as `window.visible` would be true in these cases.

Reviewer: I agree it would be exceptionally difficult to get into this state, but it is *possible* if you invoke system native APIs. 

At the very least, it's useful for completeness/symmetry in the docs.
